crates/router/src/core/routing.rs

```diff
@@ -98,7 +99,20 @@ pub async fn retrieve_merchant_routing_dictionary(
             i64::from(query_params.offset.unwrap_or_default()),
         )
         .await
-        .to_not_found_response(errors::ApiErrorResponse::ResourceIdNotFound)?;
+        .to_not_found_response(errors::ApiErrorResponse::ResourceIdNotFound)?
+        .into_iter()
+        .filter_map(|routing_metadata| {
+            if authentication_profile_id
+                .clone()
+                .is_some_and(|auth_profile| auth_profile == routing_metadata.profile_id)
+            {
+                Some(routing_metadata)
+            } else {
+                None
+            }
+        })
+        .collect();
```

Comment:
Please check if you can use this.
https://github.com/juspay/hyperswitch/blob/379831932a9c1e73b669f40af75cd02e682a0ec7/crates/router/src/core/payments.rs#L2935